app/models/ElasticRetriever.scala

```diff
@@ -183,6 +183,68 @@ class ElasticRetriever @Inject()(client: ElasticClient,
         (mappedHits, aggs)
     }
 
+  def getQ[A](esIndex: String,
+              boolQ: BoolQuery,
+              pageSize: Int,
+              buildF: JsValue => Option[A],
+              aggs: Iterable[AbstractAggregation] = Iterable.empty,
+              sortByFields: List[sort.FieldSort] = Nil,
+              excludedFields: Seq[String] = Seq.empty,
+              searchAfter: Option[String] = None): Future[(IndexedSeq[A], Long, Option[String])] = {
+
+    val sa = decodeSearchAfter(searchAfter)
+    val q = search(esIndex)
+      .bool(boolQ)
+      .size(pageSize)
+      .aggs(aggs)
+      .trackTotalHits(true)
+      .sourceExclude(excludedFields)
+      .searchAfter(sa)
+
+    // just log and execute the query
+    val elems: Future[Response[SearchResponse]] = client.execute {
+      val qq = sortByFields match {
+        case Nil => q
+        case _ =>
+          q.sortBy(sortByFields:_*)
+
+      }
+
+      logger.debug(client.show(qq))
+      qq
+    }
+
+    elems.map {
+      case _: RequestFailure                       => (IndexedSeq.empty, 0, None)
+      case results: RequestSuccess[SearchResponse] =>
+        // parse the full body response into JsValue
+        // thus, we can apply Json Transformations from JSON Play
+        val result = Json.parse(results.body.get)
+
+        logger.debug(Json.prettyPrint(result))
+        val hits = (result \ "hits" \ "hits").get.as[JsArray].value
+        val totalHits = results.result.totalHits
+
+        val mappedHits = hits
+          .map(jObj => {
+            buildF(jObj)
+          })
+          .withFilter(_.isDefined)
+          .map(_.get)
+
+        val hasNext = !(hits.size < pageSize) && pageSize > 0
```

Comment:
I have a feeling that I might have written this line, but looking at it now it really should be 
``` 
val hasNext = pageSize < hits.size && pageSize > 0
```